internal/handlers/pg/msg_renamecollection.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pg
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/jackc/pgx/v5"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/handlers/pg/pgdb"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// MsgRenameCollection implements HandlerInterface.
+func (h *Handler) MsgRenameCollection(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
+	var err error
+
+	dbPool, err := h.DBPool(ctx)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if err = common.Unimplemented(document, "writeConcern", "comment", "dropTarget"); err != nil {
+		return nil, err
+	}
+
+	sourceNamespace, err := common.GetRequiredParam[string](document, document.Command())
+	if err != nil {
+		return nil, err
+	}
+
+	sourceDB, collection, err := extractFromNamespace(sourceNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	targetField, err := document.Get("to")
+	if err != nil {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrMissingField,
+			"BSON field 'renameCollection.to' is missing but a required field",
+			document.Command(),
+		)
+	}
+
+	targetNamespace, ok := targetField.(string)
+	if !ok {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrTypeMismatch,
+			"BSON field 'renameCollection.to' is the wrong type 'bool', expected type 'string'",
+			document.Command(),
+		)
+	}
+
+	// we assume we cannot move a collection between databases, yet.
+	targetDB, targetCollection, err := extractFromNamespace(targetNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	if targetDB != sourceDB {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrNotImplemented,
+			"Can't rename a collection to another database",
```

Comment:
```suggestion
			"Can't rename a collection in another database",
```

Reply:
This message makes users believe it's an invalid operation instead of not implemented. It should be more consistent with our other unimplemented error message. Something like `'renameCollection' in another database is not implemented`.